internal/backend/sessions/sessioncalls/call.go

```diff
@@ -66,12 +67,24 @@ func (cs *calls) invoke(ctx context.Context, callv sdktypes.Value, args []sdktyp
 		}
 	}
 
-	v, err := caller.Call(ctx, callv, args, kwargs)
+	v, err := func() (v sdktypes.Value, err error) {
+		defer func() {
+			if r := recover(); r != nil {
+				err = fmt.Errorf("panic: %v", r)
+			}
+		}()
+
+		return caller.Call(ctx, callv, args, kwargs)
+	}()
 	if err != nil {
 		if errors.Is(err, workflow.ErrCanceled) && errors.Is(ctx.Err(), context.DeadlineExceeded) {
 			err = context.DeadlineExceeded
 		}
 
+		if errors.Is(err, test.ErrTestHard) {
```

Comment:
Adding test-only code isn't strictly wrong, sometimes it's a valid choice, and I agree with it in this case.

But in general it is considered an anti-pattern because you're adding dead code to production, and risking having tests that test themselves and not the product.

So a short comment about the purpose of this logical branch, and maybe a reference to the test(s) that use it may be good ideas.